library/common/extensions/filters/http/platform_bridge/filter.cc

```diff
@@ -321,6 +322,36 @@ Http::FilterHeadersStatus PlatformBridgeFilter::encodeHeaders(Http::ResponseHead
                                                               bool end_stream) {
   ENVOY_LOG(trace, "PlatformBridgeFilter({})::encodeHeaders", filter_name_);
 
+  // Presence of internal error header indicates an error that should be surfaced as an
+  // error callback (rather than an HTTP response).
+  const auto error_code_header = headers.get(Http::InternalHeaders::get().ErrorCode);
```

Comment:
Are we able to have a test case for verifying this behavior?

Reply:
Added passing test.